src/main.rs

```diff
@@ -407,8 +500,60 @@ unsafe extern "C" fn roc_fx_createCamera(
     match alloc_result {
         Ok(roc_box) => RocResult::ok(roc_box),
         Err(_) => {
-            exit_with_msg("Unable to load camera, out of memory in the camera heap. Consider using ROC_RAY_MAX_CAMERAS_HEAP_SIZE env var to increase the heap size.".into());
-            std::process::exit(1);
+            exit_with_msg("Unable to load camera, out of memory in the camera heap. Consider using ROC_RAY_MAX_CAMERAS_HEAP_SIZE env var to increase the heap size.".into(), ExitErrCode::ExitHeapFull);
+        }
+    }
+}
+
+#[no_mangle]
+unsafe extern "C" fn roc_fx_createRenderTexture(
+    size: &glue::RocVector2,
+) -> RocResult<RocBox<()>, ()> {
+    if !is_effect_permitted(PlatformEffect::CreateRenderTexture) {
+        let mode = platform_mode_str();
+        exit_with_msg(
+            format!("Cannot create a render texture while in {mode}"),
+            ExitErrCode::ExitEffectNotPermitted,
+        );
+    }
+
+    let (width, height) = size.to_components_c_int();
+
+    // [src/main.rs:521:5] width = 400
+    // [src/main.rs:521:5] height = 300
+    dbg!(width, height);
+
+    let render_texture: bindings::RenderTexture = bindings::LoadRenderTexture(width, height);
+
+    // [src/main.rs:524:5] &render_texture = RenderTexture {
+    //     id: 1,
+    //     texture: Texture {
+    //         id: 3,
+    //         width: 400,
+    //         height: 300,
+    //         mipmaps: 1,
+    //         format: 7,
+    //     },
+    //     depth: Texture {
+    //         id: 1,
+    //         width: 400,
+    //         height: 300,
+    //         mipmaps: 1,
+    //         format: 19,
+    //     },
+    // }
+    dbg!(&render_texture);
+
+    let heap = roc::render_texture_heap();
+
+    // WHY IS THIS FAILING??
+    // roc-ray(54393,0x2015c0f40) malloc: *** error for object 0x10ba68000: pointer being freed was not allocated
+    // roc-ray(54393,0x2015c0f40) malloc: *** set a breakpoint in malloc_error_break to debug
```

Comment:
@bhansconnect I think I need your help to investigate this issue. ASFAIK I am doing everything correct. I'm trying to add support for `RenderTexture`. [here is an official raylib example using this feature](https://github.com/raysan5/raylib/blob/master/examples/core/core_2d_camera_split_screen.c)

This is fourth heap I'm adding and I've done it the same as the previous ones... so I feel like this should work.

Repro with `$ just dev`